checkov/terraform/runner.py

```diff
@@ -414,17 +416,38 @@ def _strip_module_referrer(file_path: str) -> Tuple[str, Optional[str]]:
         else:
             return file_path, None
 
-    @staticmethod
-    def _find_id_for_referrer(full_file_path, definitions) -> Optional[str]:
-        for file, file_content in definitions.items():
-            if "module" not in file_content:
-                continue
+    def _find_id_for_referrer(self, full_file_path) -> Optional[str]:
+        cached_referrer = self.referrer_cache.get(full_file_path)
+        if cached_referrer:
+            return cached_referrer
+        if not self.definitions_with_modules:
+            self._prepare_definitions_with_modules()
+        for file, file_content in self.definitions_with_modules.items():
 
             for modules in file_content["module"]:
                 for module_name, module_content in modules.items():
                     if "__resolved__" not in module_content:
                         continue
 
                     if full_file_path in module_content["__resolved__"]:
-                        return f"module.{module_name}"
+                        id_referrer = f"module.{module_name}"
+                        self.referrer_cache[full_file_path] = id_referrer
+                        return id_referrer
         return None
+
+    def _prepare_definitions_with_modules(self):
+        for file, file_content in self.definitions.items():
+            if "module" not in file_content:
+                continue
+
+            for modules in file_content["module"]:
+                cached = False
+                for module_name, module_content in modules.items():
```

Comment:
thought about this double `for` loop with two `break`, you could rewrite it in two ways.
1. move it a separate function, remove both `break` and instead just use ` return file_content`
2. leverage `for` ... `else` ... this one looks always strange to me, but would be correct Python code 😄